psychopy/experiment/components/keyboard/__init__.py

```diff
@@ -225,49 +232,46 @@ def writeFrameCode(self, buff):
             keyListStr = "keyList=%s" % repr(keyList)
 
         # check for keypresses
-        buff.writeIndented("theseKeys = event.getKeys(%s)\n" % keyListStr)
+        code = "theseKeys = %s.getKeys(%s, waitRelease=False)\n" % (self.params['name'], keyListStr)
+        buff.writeIndented(code)
+
+        # Check for response
+        buff.writeIndented("if len(theseKeys):\n")
+        buff.setIndentLevel(1, True)
+        dedentAtEnd += 1  # indent by 1
+        buff.writeIndented("theseKeys = theseKeys[0]  # at least one key was pressed\n")
 
         if self.exp.settings.params['Enable Escape'].val:
             code = ('\n# check for quit:\n'
-                    'if "escape" in theseKeys:\n'
+                    'if "escape" in theseKeys.name:\n'
```

Comment:
Actually... I wrote it so that this could be a comparison `if "escape" == theseKeys:` (without you needing to include `.name`) although it looks like I need to do something similar for `includes` operations

Reply:
Ah, yes that should actually be a equality operator. I had written it at the time in case the name attribute was also a list. Equality check will work for both types of object returned PTB and event modules.

Reply:
Fixed in #2373